### DISCUSSION TITLE
Lesson 6: private key safety pt.2 - source .env

### DISCUSSION THREAD

**Discussion by danker-dotcom** [Cyfrin/foundry-full-course-cu#1672] [Q&A] [answered]
When entering `source .env` into the terminal, I am returned the following : 
```
PRIVATE_KEY: command not found
RPC_URL: command not found
```
This is the code I have in my `.env` file: 
```
PRIVATE_KEY = 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
RPC_URL = http://127.0.0.1:8545
```
I'm guessing this next code wouldn't run w/o getting the previous to work, but ill share the error output anyway: 
`forge script script/DeploySimpleStorage.s.sol --rpc-url $RPC_URL --broadcast --private-key $PRIVATE_KEY`
output:
`error: a value is required for '--fork-url <URL>' but none was supplied`

`
`
When running the codes `echo $PRIVATE_KEY `, and `echo RPC_URL` in the terminal, I'm given a blank output. 
Also, I am running anvil in the other terminal while i do this. 
I was wondering if I have some kind of security setting on in VS Code that is preventing it from being accessed.

**Comment by EngrPips** [comment]
Hello @danker-dotcom, I want to mention that you should not get into the habit of showing your private key publicly. Is your `.env` file at the top level of your working directory? 

Please show your working directory as I  show mine below 

<img width="1430" alt="Screenshot 2024-05-03 at 12 53 59" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/f3ab9998-2831-4bd7-9fd5-657b91e10f6a">

**Comment by danker-dotcom** [comment]
Here is my working directory: 

![foundry-SS-chp6 16](https://github.com/Cyfrin/foundry-full-course-f23/assets/155930955/815f571d-66ca-4fd7-b3ca-95d4a52e317b)

**Reply by EngrPips** [reply, marked as answer]
Thanks, @danker-dotcom. Correct your code to the below in your `.env` file, and things should start working fine.

```javascript
PRIVATE_KEY=0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
RPC_URL=http://127.0.0.1:8545
```
Don't put any space between your `variables`, `=` and `values`.

**Comment by danker-dotcom** [comment]
Oh shoot my mistake, thank you, it now works. 

Also, is there a reason for the `echo $PRIVATE_KEY` and `echo RPC_URL` to fetch blank/invisible outputs. A security setting perhaps?

**Reply by EngrPips** [reply]
Did it still fetch blank after your corrected that mistake?

**Reply by danker-dotcom** [reply]
You know what, it didn't at first. But I had restarted VS and it then worked. Thanks lots.

**Reply by EngrPips** [reply]
Anytime, Friend. Keep crushing it.

**Reply by dennnis0204** [reply]
Must be 'source .env' in CLI before 'echo $PRIVATE_KEY' in my case

**Reply by EngrPips** [reply]
Yeah that is logical.
